### ISSUE TITLE
[IOS-XE] Show Version does not parser SMUs

### ISSUE THREAD

**Issue by dmychepk** [CiscoTestAutomation/genieparser#871]
2024-07-17 17:24:54,462: %UNICON-INFO: +++ svs-fc-agg-t1-a-1 with via 'ssh': executing command 'show version' +++
show version
Cisco IOS XE Software, Version 17.09.04a - SMU-PATCHED
Cisco IOS Software [Cupertino], Catalyst L3 Switch Software (CAT9K_IOSXE), Version 17.9.4a, RELEASE SOFTWARE (fc3)
Technical Support: http://www.cisco.com/techsupport
Copyright (c) 1986-2023 by Cisco Systems, Inc.
Compiled Fri 20-Oct-23 10:44 by mcpre

Cisco IOS-XE software, Copyright (c) 2005-2023 by cisco Systems, Inc.
All rights reserved.  Certain components of Cisco IOS-XE software are
licensed under the GNU General Public License ("GPL") Version 2.0.  The
software code licensed under GPL Version 2.0 is free software that comes
with ABSOLUTELY NO WARRANTY.  You can redistribute and/or modify such
GPL code under the terms of GPL Version 2.0.  For more details, see the
documentation or "License Notice" file accompanying the IOS-XE software,
or the applicable URL provided on the flyer accompanying the IOS-XE
software.

ROM: IOS-XE ROMMON
BOOTLDR: System Bootstrap, Version 17.8.1r[FC1], RELEASE SOFTWARE (P)

svs-fc-agg-t1-a-1 uptime is 5 days, 5 hours, 14 minutes
Uptime for this control processor is 5 days, 5 hours, 16 minutes
System returned to ROM by Reload Command at 05:07:25 PDT Fri Jul 12 2024
System restarted at 08:10:08 EDT Fri Jul 12 2024
System image file is "bootflash:packages.conf"
Last reload reason: Reload Command

This product contains cryptographic features and is subject to United
States and local country laws governing import, export, transfer and
use. Delivery of Cisco cryptographic products does not imply
third-party authority to import, export, distribute or use encryption.
Importers, exporters, distributors and users are responsible for
compliance with U.S. and local country laws. By using this product you
agree to comply with applicable laws and regulations. If you are unable
to comply with U.S. and local laws, return this product immediately.

A summary of U.S. laws governing Cisco cryptographic products may be found at:
http://www.cisco.com/wwl/export/crypto/tool/stqrg.html

If you require further assistance please contact us by sending email to
export@cisco.com.

Technology Package License Information: 

------------------------------------------------------------------------------
Technology-package                                     Technology-package
Current                        Type                       Next reboot  
------------------------------------------------------------------------------
network-advantage   	Smart License                 	 network-advantage   
dna-advantage       	Subscription Smart License    	 dna-advantage                 
AIR License Level: AIR DNA Advantage
Next reload AIR license Level: AIR DNA Advantage

_Active SMU Information: 
  State (St):  C - Committed, U - Uncommitted
--------------------------------------------------------------------------------
Type  Defect_ID    Version   St Filename                                     
--------------------------------------------------------------------------------
SMU   CSCwi56960   17.09.04a C  cat9k_iosxe.17.09.04a.CSCwi56960.SPA.smu.bin
--------------------------------------------------------------------------------_

Smart Licensing Status: Smart Licensing Using Policy

cisco C9500-48Y4C (X86) processor with 2883506K/6147K bytes of memory.
Processor board ID FDO270903SJ
1 Virtual Ethernet interface
48 TwentyFive Gigabit Ethernet interfaces
4 Hundred Gigabit Ethernet interfaces
32768K bytes of non-volatile configuration memory.
15993664K bytes of physical memory.
11161600K bytes of Bootflash at bootflash:.
1638400K bytes of Crash Files at crashinfo:.
234430023K bytes of SATA hard disk at disk0:.

Base Ethernet MAC Address          : 74:8f:c2:e3:fc:80
Motherboard Assembly Number        : 4873
Motherboard Serial Number          : FDO270903SJ
Model Revision Number              : V02
Motherboard Revision Number        : 5
Model Number                       : C9500-48Y4C
System Serial Number               : FDO270903SJ

{'version': {'air_license_level': 'AIR DNA Advantage',
             'bootldr': 'System Bootstrap, Version 17.8.1r[FC1], RELEASE '
                        'SOFTWARE (P)',
             'chassis': 'C9500-48Y4C',
             'chassis_sn': 'FDO270903SJ',
             'compiled_by': 'mcpre',
             'compiled_date': 'Fri 20-Oct-23 10:44',
             'copyright_years': '1986-2023',
             'disks': {'bootflash:.': {'disk_size': '11161600',
                                       'type_of_disk': 'Bootflash'},
                       'crashinfo:.': {'disk_size': '1638400',
                                       'type_of_disk': 'Crash Files'},
                       'disk0:.': {'disk_size': '234430023',
                                   'type_of_disk': 'SATA hard disk'}},
             'hostname': 'svs-fc-agg-t1-a-1',
             'image_id': 'CAT9K_IOSXE',
             'image_type': 'production image',
             'label': 'RELEASE SOFTWARE (fc3)',
             'last_reload_reason': 'Reload Command',
             'license_package': {'dna-advantage': {'license_level': 'dna-advantage',
                                                   'license_type': 'Subscription '
                                                                   'Smart '
                                                                   'License',
                                                   'next_reload_license_level': 'dna-advantage'},
                                 'network-advantage': {'license_level': 'network-advantage',
                                                       'license_type': 'Smart '
                                                                       'License',
                                                       'next_reload_license_level': 'network-advantage'}},
             'location': 'Cupertino',
             'main_mem': '2883506',
             'mem_size': {'non-volatile configuration': '32768',
                          'physical': '15993664'},
             'next_reload_air_license_level': 'AIR DNA Advantage',
             'number_of_intfs': {'Hundred Gigabit Ethernet': '4',
                                 'TwentyFive Gigabit Ethernet': '48',
                                 'Virtual Ethernet': '1'},
             'os': 'IOS-XE',
             'platform': 'Catalyst L3 Switch',
             'processor_type': 'X86',
             'returned_to_rom_at': '05:07:25 PDT Fri Jul 12 2024',
             'returned_to_rom_by': 'Reload Command',
             'rom': 'IOS-XE ROMMON',
             'rtr_type': 'C9500-48Y4C',
             'switch_num': {'1': {'mac_address': '74:8f:c2:e3:fc:80',
                                  'mb_assembly_num': '4873',
                                  'mb_rev_num': '5',
                                  'mb_sn': 'FDO270903SJ',
                                  'model_num': 'C9500-48Y4C',
                                  'model_rev_num': 'V02',
                                  'system_sn': 'FDO270903SJ'}},
             'system_image': 'bootflash:packages.conf',
             'system_restarted_at': '08:10:08 EDT Fri Jul 12 2024',
             'uptime': '5 days, 5 hours, 14 minutes',
             'uptime_this_cp': '5 days, 5 hours, 16 minutes',
             'version': '17.9.4a',
             'version_short': '17.9',
             'xe_version': '17.09.04a - SMU-PATCHED'}}

**Comment by Harishv01** [comment]
Hi, Can you please share the OS, platform, and model you are using? Also, is this  `SMU CSCwi56960 17.09.04a C cat9k_iosxe.17.09.04a.CSCwi56960.SPA.smu.bin` missing in the output?

**Comment by dmychepk** [comment]
Catalyst 9300 or 9500 with any SMU installed

Отправлено с мобильного устройства


пн, 22 лип. 2024 р. о 12:38 пп Harishv ***@***.***> пише:

**Comment by Harishv01** [comment]
We have multiple parsers for the show version. Can you please share the exact details you mentioned in the testbed i.e OS, platform, and model?

**Comment by dmychepk** [comment]
You can see all details in my show version output I attached to case

Отправлено с мобильного устройства


пн, 22 лип. 2024 р. о 1:00 пп Harishv ***@***.***> пише:

**Comment by Harishv01** [comment]
Hi, I have fixed this and the output looks as follows below. Please review it and let me know if anything is missing. If everything looks good, I will raise a pull request and merge it. Thank you


```
expected_output ={
    "version": {
        "xe_version": "17.09.04a - SMU-PATCHED",
        "version_short": "17.9",
        "code_name": "Cupertino",
        "platform": "Catalyst L3 Switch",
        "image_id": "CAT9K_IOSXE",
        "version": "17.9.4",
        "label": "a, RELEASE SOFTWARE (fc3)",
        "compiled_date": "Fri 20-Oct-23 10:44",
        "compiled_by": "mcpre",
        "os": "IOS-XE",
        "rom": "IOS-XE ROMMON",
        "bootldr_version": "System Bootstrap, Version 17.8.1r[FC1], RELEASE SOFTWARE (P)",
        "hostname": "svs-fc-agg-t1-a-1",
        "uptime": "5 days, 5 hours, 14 minutes",
        "uptime_this_cp": "5 days, 5 hours, 16 minutes",
        "returned_to_rom_by": "Reload Command at 05:07:25 PDT Fri Jul 12 2024",
        "system_image": "bootflash:packages.conf",
        "last_reload_reason": "Reload Command",
        "license_level": "AIR DNA Advantage",
        "next_reload_license_level": "AIR DNA Advantage",
        "type": "SMU",
        "defect_id": "CSCwi56960",
        "sum_version": "17.09.04a",
        "state": "C",
        "file_name": "cat9k_iosxe.17.09.04a.CSCwi56960.SPA.smu.bin",
        "smart_licensing_status": "Smart Licensing Using Policy",
        "chassis": "C9500-48Y4C",
        "processor_type": "X86",
        "main_mem": "2883506",
        "processor_board_id": "FDO270903SJ",
        "number_of_intfs": {
            "hundred_gigabit_ethernet_interfaces": "4"
        },
        "mem_size": {
            "non_volatile_memory": "32768",
            "physical_memory": "15993664"
        },
        "disks": {
            "bootflash:": {
                "disk_size": "11161600"
            },
            "crashinfo:": {
                "disk_size": "1638400"
            }
        },
        "mac_address": "74:8f:c2:e3:fc:80",
        "mb_assembly_num": "4873",
        "mb_sn": "FDO270903SJ",
        "model_rev_num": "V02",
        "mb_rev_num": "5",
        "model_num": "C9500-48Y4C",
        "system_sn": "FDO270903SJ"
    }
}
```

**Comment by dmychepk** [comment]
There could be multiple SMUs on devices, so it should have some keyword
like smu and have availability to parse multiple

Отправлено с мобильного устройства


ср, 24 лип. 2024 р. о 10:10 дп Harishv ***@***.***> пише:

**Comment by Harishv01** [comment]
Could you please provide an example so that I can fix it accordingly?

**Comment by Harishv01** [comment]
Hi,Could you please provide an example so that I can fix it accordingly?Thank you

**Comment by dmychepk** [comment]
2024-03-06 13:31:39,357: %UNICON-INFO: +++ svs-fc-dis-r-a-1 with via 'ssh': executing command 'show version' +++
show version
Cisco IOS XE Software, Version 17.03.03 - SMU-PATCHED
Cisco IOS Software [Amsterdam], Catalyst L3 Switch Software (CAT9K_IOSXE), Version 17.3.3, RELEASE SOFTWARE (fc7)
Technical Support: http://www.cisco.com/techsupport
Copyright (c) 1986-2021 by Cisco Systems, Inc.
Compiled Thu 04-Mar-21 12:32 by mcpre


Cisco IOS-XE software, Copyright (c) 2005-2021 by cisco Systems, Inc.
All rights reserved.  Certain components of Cisco IOS-XE software are
licensed under the GNU General Public License ("GPL") Version 2.0.  The
software code licensed under GPL Version 2.0 is free software that comes
with ABSOLUTELY NO WARRANTY.  You can redistribute and/or modify such
GPL code under the terms of GPL Version 2.0.  For more details, see the
documentation or "License Notice" file accompanying the IOS-XE software,
or the applicable URL provided on the flyer accompanying the IOS-XE
software.


ROM: IOS-XE ROMMON
BOOTLDR: System Bootstrap, Version 17.8.1r[FC1], RELEASE SOFTWARE (P)

svs-fc-dis-r-a-1 uptime is 11 hours, 42 minutes
Uptime for this control processor is 11 hours, 44 minutes
System returned to ROM by Install  at 09:32:14 PST Wed Feb 21 2024
System restarted at 01:48:52 UTC Wed Mar 6 2024
System image file is "bootflash:packages.conf"
Last reload reason: Install 



This product contains cryptographic features and is subject to United
States and local country laws governing import, export, transfer and
use. Delivery of Cisco cryptographic products does not imply
third-party authority to import, export, distribute or use encryption.
Importers, exporters, distributors and users are responsible for
compliance with U.S. and local country laws. By using this product you
agree to comply with applicable laws and regulations. If you are unable
to comply with U.S. and local laws, return this product immediately.

A summary of U.S. laws governing Cisco cryptographic products may be found at:
http://www.cisco.com/wwl/export/crypto/tool/stqrg.html

If you require further assistance please contact us by sending email to
export@cisco.com.


Technology Package License Information: 

------------------------------------------------------------------------------
Technology-package                                     Technology-package
Current                        Type                       Next reboot  
------------------------------------------------------------------------------
network-advantage   	Smart License                 	 network-advantage   
dna-advantage       	Subscription Smart License    	 dna-advantage                 
AIR License Level: AIR DNA Advantage
Next reload AIR license Level: AIR DNA Advantage

Active SMU Information: 
  State (St):  C - Committed, U - Uncommitted
--------------------------------------------------------------------------------
Type  Defect_ID    Version   St Filename                                     
--------------------------------------------------------------------------------
SMU   CSCvz54210   17.03.03. C  cat9k_iosxe.17.03.03.CSCvz54210.SPA.smu.bin  
SMU   CSCwc80043   17.03.03. C  cat9k_iosxe.17.03.03.CSCwc80043.SPA.smu.bin  
SMU   CSCwh29778   17.03.03. C  cat9k_iosxe.17.03.03.CSCwh29778.SPA.smu.bin  
--------------------------------------------------------------------------------

Smart Licensing Status: Registration Not Applicable/Not Applicable

cisco C9500-48Y4C (X86) processor with 2904385K/6147K bytes of memory.
Processor board ID FDO25180ANN
55 Virtual Ethernet interfaces
96 TwentyFive Gigabit Ethernet interfaces
8 Hundred Gigabit Ethernet interfaces
32768K bytes of non-volatile configuration memory.
16002876K bytes of physical memory.
11161600K bytes of Bootflash at bootflash:.
11161600K bytes of Bootflash at bootflash-2:.
1638400K bytes of Crash Files at crashinfo:.
1638400K bytes of Crash Files at crashinfo-2:.

Base Ethernet MAC Address          : 34:73:2d:42:e8:a0
Motherboard Assembly Number        : 4873
Motherboard Serial Number          : FDO25180ANN
Model Revision Number              : V02
Motherboard Revision Number        : 5
Model Number                       : C9500-48Y4C         
System Serial Number               : FDO25180ANN

Switch 02
---------
Base Ethernet MAC Address          : 4c:5d:3c:53:49:00
Motherboard Assembly Number        : 4873
Motherboard Serial Number          : FDO251801UK
Model Revision Number              : V02
Motherboard Revision Number        : 5
Model Number                       : C9500-48Y4C         
System Serial Number               : FDO251801UK

**Comment by Harishv01** [comment]
Hi, I have fixed this and the output looks as follows below. Please review it and let me know if anything is missing. If everything looks good, I will raise a pull request and merge it. Thank you

```
expected_output ={
    "version": {
        "xe_version": "17.03.03 - SMU-PATCHED",
        "version_short": "17.3",
        "code_name": "Amsterdam",
        "platform": "Catalyst L3 Switch",
        "image_id": "CAT9K_IOSXE",
        "version": "17.3.3",
        "label": "RELEASE SOFTWARE (fc7)",
        "compiled_date": "Thu 04-Mar-21 12:32",
        "compiled_by": "mcpre",
        "os": "IOS-XE",
        "rom": "IOS-XE ROMMON",
        "bootldr_version": "System Bootstrap, Version 17.8.1r[FC1], RELEASE SOFTWARE (P)",
        "hostname": "svs-fc-dis-r-a-1",
        "uptime": "11 hours, 42 minutes",
        "uptime_this_cp": "11 hours, 44 minutes",
        "returned_to_rom_by": "Install  at 09:32:14 PST Wed Feb 21 2024",
        "system_image": "bootflash:packages.conf",
        "last_reload_reason": "Install",
        "license_level": "AIR DNA Advantage",
        "next_reload_license_level": "AIR DNA Advantage",
        "defect_id": {
            "CSCvz54210": {
                "type": "SMU",
                "sum_version": "17.03.03.",
                "state": "C",
                "file_name": "cat9k_iosxe.17.03.03.CSCvz54210.SPA.smu.bin"
            },
            "CSCwc80043": {
                "type": "SMU",
                "sum_version": "17.03.03.",
                "state": "C",
                "file_name": "cat9k_iosxe.17.03.03.CSCwc80043.SPA.smu.bin"
            },
            "CSCwh29778": {
                "type": "SMU",
                "sum_version": "17.03.03.",
                "state": "C",
                "file_name": "cat9k_iosxe.17.03.03.CSCwh29778.SPA.smu.bin"
            }
        },
        "smart_licensing_status": "Registration Not Applicable/Not Applicable",
        "chassis": "C9500-48Y4C",
        "processor_type": "X86",
        "main_mem": "2904385",
        "processor_board_id": "FDO25180ANN",
        "number_of_intfs": {
            "virtual_ethernet_interfaces": "55",
            "hundred_gigabit_ethernet_interfaces": "8"
        },
        "mem_size": {
            "non_volatile_memory": "32768",
            "physical_memory": "16002876"
        },
        "disks": {
            "bootflash:": {
                "disk_size": "11161600"
            },
            "crashinfo:": {
                "disk_size": "1638400"
            }
        },
        "mac_address": "4c:5d:3c:53:49:00",
        "mb_assembly_num": "4873",
        "mb_sn": "FDO251801UK",
        "model_rev_num": "V02",
        "mb_rev_num": "5",
        "model_num": "C9500-48Y4C",
        "system_sn": "FDO251801UK"
    }
}
```

**Comment by Harishv01** [comment]
Hi, I have fixed this and the output looks as follows below. Please review it and let me know if anything is missing. If everything looks good, I will raise a pull request and merge it. Thank you

```
expected_output ={
    "version": {
        "xe_version": "17.03.03 - SMU-PATCHED",
        "version_short": "17.3",
        "code_name": "Amsterdam",
        "platform": "Catalyst L3 Switch",
        "image_id": "CAT9K_IOSXE",
        "version": "17.3.3",
        "label": "RELEASE SOFTWARE (fc7)",
        "compiled_date": "Thu 04-Mar-21 12:32",
        "compiled_by": "mcpre",
        "os": "IOS-XE",
        "rom": "IOS-XE ROMMON",
        "bootldr_version": "System Bootstrap, Version 17.8.1r[FC1], RELEASE SOFTWARE (P)",
        "hostname": "svs-fc-dis-r-a-1",
        "uptime": "11 hours, 42 minutes",
        "uptime_this_cp": "11 hours, 44 minutes",
        "returned_to_rom_by": "Install  at 09:32:14 PST Wed Feb 21 2024",
        "system_image": "bootflash:packages.conf",
        "last_reload_reason": "Install",
        "license_level": "AIR DNA Advantage",
        "next_reload_license_level": "AIR DNA Advantage",
        "defect_id": {
            "CSCvz54210": {
                "type": "SMU",
                "sum_version": "17.03.03.",
                "state": "C",
                "file_name": "cat9k_iosxe.17.03.03.CSCvz54210.SPA.smu.bin"
            },
            "CSCwc80043": {
                "type": "SMU",
                "sum_version": "17.03.03.",
                "state": "C",
                "file_name": "cat9k_iosxe.17.03.03.CSCwc80043.SPA.smu.bin"
            },
            "CSCwh29778": {
                "type": "SMU",
                "sum_version": "17.03.03.",
                "state": "C",
                "file_name": "cat9k_iosxe.17.03.03.CSCwh29778.SPA.smu.bin"
            }
        },
        "smart_licensing_status": "Registration Not Applicable/Not Applicable",
        "chassis": "C9500-48Y4C",
        "processor_type": "X86",
        "main_mem": "2904385",
        "processor_board_id": "FDO25180ANN",
        "number_of_intfs": {
            "virtual_ethernet_interfaces": "55",
            "hundred_gigabit_ethernet_interfaces": "8"
        },
        "mem_size": {
            "non_volatile_memory": "32768",
            "physical_memory": "16002876"
        },
        "disks": {
            "bootflash:": {
                "disk_size": "11161600"
            },
            "crashinfo:": {
                "disk_size": "1638400"
            }
        },
        "mac_address": "4c:5d:3c:53:49:00",
        "mb_assembly_num": "4873",
        "mb_sn": "FDO251801UK",
        "model_rev_num": "V02",
        "mb_rev_num": "5",
        "model_num": "C9500-48Y4C",
        "system_sn": "FDO251801UK"
    }
```
}

**Comment by Harishv01** [comment]
Hi, We have merged the PR, and the fix will be available in the next version, which is 24.8. Hence, I am closing the ticket. Thank you.
